controllers/containerimagebuild_controller.go

```diff
@@ -149,56 +141,65 @@ func (r *ContainerImageBuildReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
-func (r *ContainerImageBuildReconciler) getAuthCredentials(ctx context.Context, log logr.Logger, registry forgev1alpha1.Registry) (username, password string, err error) {
-	switch registry.BasicAuth {
-	case "":
-		log.Info("Not using registry credentials")
-		return
-	case forgev1alpha1.BasicAuthInline:
-		log.Info("Processing inline registry credentials")
-		username = registry.Username
-		password = registry.Password
-	case forgev1alpha1.BasicAuthSecret:
-		log.Info("Processing secret registry credentials")
-		var secret corev1.Secret
-		if err = r.Client.Get(ctx, types.NamespacedName{
-			Namespace: registry.SecretNamespace,
-			Name:      registry.SecretName,
-		}, &secret); err != nil {
-			err = fmt.Errorf("cannot find registry auth secret: %v", err)
-			return
-		}
+// NOTE: the following 2 funcs should not really be attached to the Reconciler
```

Comment:
Both seem fairly stateless, why do they need to be in the Reconciler for now vs. just a function in the same package?

Reply:
Oh just for `Client`?

Reply:
Yup. In another PR I'll remove the dependency on the Reconciler client and just use the K8s client directly. Truthfully, I don't really care for this client on account of it requiring weird RBAC permissions for proper operation. It may be required for the controller but we don't need it for other K8s operations.